src/User.jsx

```diff
@@ -0,0 +1,322 @@
+
+import React, {useState, Component, useRef, useEffect} from 'react';
+import classnames from 'classnames';
+import styles from './Header.module.css';
+import * as ceramicApi from '../ceramic.js';
+// import styles from './User.module.css';
+import {storageHost, accountsHost, tokensHost, loginEndpoint, discordClientId, walletHost} from '../constants';
+import {contracts, getAddressFromMnemonic} from '../blockchain.js';
+import {jsonParse, parseQuery, handleDiscordLogin} from '../util.js';
+import Modal from "./components/modal";
+
+function useComponentVisible(initialIsVisible, fn) {
+  const [isComponentVisible, setIsComponentVisible] = useState(initialIsVisible);
+  const ref = useRef(null);
+
+  const handleClickOutside = event => {
```

Comment:
For this appear/disappear, can we use the existing infrastructure rather than adding more global event handlers?

Reply:
https://github.com/webaverse/app/blob/b06ba4455f2f4ff1dfb627c2f78f722c5b8aba48/src/Header.jsx#L494